### DISCUSSION TITLE
Which highlight group went wrong?

### DISCUSSION THREAD

**Discussion by jackielii** [neoclide/coc.nvim#4125] [Q&A]
I want to file an issue wrt the recent highlight group change, but don't know how to find where the problem is.

See this screenshot:
![image](https://user-images.githubusercontent.com/360983/187463073-bf84cb67-fc37-4dc5-afd9-bac58111b6d6.png)

There are 3 issues:

1. in the pop up menu, the currently selected item's highlight is too bright. It seems to be linked to CursorLine, but my CursorLine looks ok.
2. CocList preview (find reference in this case), the CursorLine's highlight looks broken
3. CocList list's current item should have the text `NewWorker` in yellow highlight (I believe it's the `Search` group) but appears to be overwritten by `CursorLine`?

My question is how to find out where it has gone wrong? I haven't change my config for a while, only by upgrading the the `master` of `coc.nvim`

**Comment by jackielii** [comment]
**Update**: after adding:

```
au ColorScheme * hi CocMenuSel ctermbg=237 guibg=#13354A
```

It now looks like:
![image](https://user-images.githubusercontent.com/360983/187467636-a34365dd-b655-46bd-8c5f-3f50b3108701.png)


My highlight related configs:

```
au ColorScheme * hi MatchParen gui=italic guibg=black guifg=NONE
au ColorScheme * hi WinSeparator guibg=none
au ColorScheme * hi CocMenuSel ctermbg=237 guibg=#13354A
au ColorScheme * hi Comment gui=italic
set pumblend=15 " transparency for popup

if filereadable(expand("~/.vimrc_background"))
  source ~/.vimrc_background
endif
```

So only 3rd issue still exists: the `NewWorker` find reference text. When not selected, it looks like:

![image](https://user-images.githubusercontent.com/360983/187468048-2d266c56-fff4-41fc-8568-3b5b24935f05.png)

**Comment by chemzqm** [comment]
> the NewWorker find reference text. When not selected, it looks like:

What's wrong with it?

**Reply by chemzqm** [reply]
The text disappeared when that line is selected?

**Reply by jackielii** [reply]
Yes, when I move to the line, the yellow text `NewWorker` disappears...

**Reply by jackielii** [reply]
I have a feeling it's a combination of `Search` and `CursorLine` group. Here is the results of these two:

```
:hi Search
Search         xxx ctermfg=0 ctermbg=11 guifg=#393939 guibg=#ffcc66
:hi CursorLine
CursorLine     xxx cterm=underline guibg=#393939
:hi CocListLine
CocListLine    xxx links to CursorLine
:hi CocSearch
CocSearch      xxx ctermfg=12 guifg=#15aabf
```

**Reply by jackielii** [reply]
Turns out it's probably my colour scheme plugin's gone wrong... I'm using https://github.com/RRethy/nvim-base16.

Setting the `CocListLine` to default `CursorLine` style works ok: 

```vim
au ColorScheme * hi CocListLine cterm=underline guibg=Grey40
```

![image](https://user-images.githubusercontent.com/360983/188005957-84333744-120b-43b7-8681-42381f8fd2b3.png)

filed https://github.com/RRethy/nvim-base16/issues/60

**Reply by jackielii** [reply]
@chemzqm I think I get it now. `CursorLine` overwritten the background of `Search`:

`Search` group in my colour scheme is already inverted: bg is Yellow, fg is the same as the CursorLine. So when `CursorLine` is applied, the bg of `Search` is removed, making the text disappear.

Is manually setting the `CocListLine` the best solution? or is there a better one? like setting the Search+CursorLine to be something else?

**Reply by chemzqm** [reply]
Do you have correct search highlight when cursor not on that line?

**Reply by jackielii** [reply]
Yes, you can see the line below the cursor line: word `ts` is highlighted correctly.

**Reply by chemzqm** [reply]
Should be the same bug https://github.com/neovim/neovim/issues/20004.
I've made the location list not use syntax highlights, but you may have other vim plugin that add syntax highlights.

**Reply by jackielii** [reply]
I see... Although I think this is the default coc list when using `<Plug>(coc-references)`.

Eitherway, thanks for the explanation, makes a lot of sense now.
